dts/bindings/timer/nxp,mcux-gpt.yaml

```diff
@@ -0,0 +1,37 @@
+#
+# Copyright (c) 2018 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+---
+title: NXP MCUX General Purpose Timer
+version: 0.1
+
+description: >
+    This is a representation of the NXP MCUX General Purpose Timer (GPT)
+
+properties:
+    compatible:
+      type: string
+      category: required
+      description: compatible strings
+      constraint: "nxp,mcux-gpt"
```

Comment:
Let's make this `nxp,imx-gpt` instead. `mcux` is a software construct

Reply:
Hi @MaureenHelm , Done.
This can be tested with counter sample.

Reply:
> > Can you please add counter to the board yaml supported list
> 
> This still needs to be done. The counter sample and tests won't get included in CI for these boards unless you do this.

yes missed that point, done now